aptos-move/framework/aptos-framework/sources/voting.move

```diff
@@ -326,6 +423,54 @@ module aptos_framework::voting {
         option::extract(&mut proposal.execution_content)
     }
 
+    /// Resolve a single-step or a multi-step proposal with the given id.
+    /// Can only be done if there are at least as many votes as min required and
+    /// there are more yes votes than no. If either of these conditions is not met, this will revert.
+    ///
+    ///
+    /// @param voting_forum_address The address of the forum where the proposals are stored.
+    /// @param proposal_id The proposal id.
+    /// @param next_execution_hash The next execution hash if the given proposal is multi-step.
+    public fun resolve_proposal_v2<ProposalType: store>(
+        voting_forum_address: address,
+        proposal_id: u64,
+        next_execution_hash: vector<u8>,
+    ) acquires VotingForum {
+        is_proposal_resolvable<ProposalType>(voting_forum_address, proposal_id);
+
+        let voting_forum = borrow_global_mut<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::borrow_mut(&mut voting_forum.proposals, proposal_id);
+
+        let multi_step_key = utf8(IS_MULTI_STEP_PROPOSAL_KEY);
+        let is_multi_step = simple_map::contains_key(&proposal.metadata, &multi_step_key) && from_bcs::to_bool(*simple_map::borrow(&proposal.metadata, &multi_step_key));
+
+        // If the proposal is a single-step proposal, or if this step is the last step in a multi-step proposal,
+        // we can mark that this proposal is resolved and emit a `ResolveProposal` event.
+        if (!is_multi_step || vector::length(&next_execution_hash) == 0) {
+            let resolved_early = can_be_resolved_early(proposal);
+            proposal.is_resolved = true;
+            proposal.resolution_time_secs = timestamp::now_seconds();
+
+            event::emit_event<ResolveProposal>(
+                &mut voting_forum.events.resolve_proposal_events,
+                ResolveProposal {
+                    proposal_id,
+                    yes_votes: proposal.yes_votes,
+                    no_votes: proposal.no_votes,
+                    resolved_early,
+                },
+            );
+
+            if (is_multi_step) {
```

Comment:
This is missing the body?

Reply:
good catch - just updated : ) I was using this if statement to set the `IS_MULTI_STEP_PROPOSAL_IN_EXECUTION` field back to `false` upon successful resolution of the multi-step proposal